### PR TITLE
add whitespace trims to tag-picker macro

### PR DESCRIPTION
**Pull request by BurningTreeC**

### CODE REVIEW

**Comment by BurningTreeC** [comment]
bump

File: core/wiki/macros/tag-picker.tid [hunk 1]
```diff
@@ -15,12 +15,13 @@ tags: $:/tags/Macro
...
 
 \define tag-picker-inner()
+\whitespace trim
 <div class="tc-edit-add-tag">
 <span class="tc-add-tag-name">
 <$keyboard key="ENTER" actions=<<add-tag-actions>>>
 <$edit-text tiddler=<<newTagNameTiddler>> tag="input" default="" placeholder={{$:/language/EditTemplate/Tags/Add/Placeholder}} focusPopup=<<qualify "$:/state/popup/tags-auto-complete">> class="tc-edit-texteditor tc-popup-handle" tabindex=<<tabIndex>> focus={{{ [{$:/config/AutoFocus}match[tags]then[true]] ~[[false]] }}}/>
 </$keyboard>
-</span> <$button popup=<<qualify "$:/state/popup/tags-auto-complete">> class="tc-btn-invisible" tooltip={{$:/language/EditTemplate/Tags/Dropdown/Hint}} aria-label={{$:/language/EditTemplate/Tags/Dropdown/Caption}}>{{$:/core/images/down-arrow}}</$button> <span class="tc-add-tag-button">
+</span>&nbsp;<$button popup=<<qualify "$:/state/popup/tags-auto-complete">> class="tc-btn-invisible" tooltip={{$:/language/EditTemplate/Tags/Dropdown/Hint}} aria-label={{$:/language/EditTemplate/Tags/Dropdown/Caption}}>{{$:/core/images/down-arrow}}</$button>&nbsp;<span class="tc-add-tag-button">
```
Comment (pmario) [inline core/wiki/macros/tag-picker.tid]: non breakable space IMO should not be used as a formatting tool. We need to use `margin-right or margin-left` here. Depending which element can or should push the other elements. ... &nbsp;'s cause problems with copy / paste actions.

Comment (Jermolene) [reply core/wiki/macros/tag-picker.tid]: This isn't really about formatting, it's just a technique for including whitespace in a tiddler with `\whitespace trim`.

**Comment by Jermolene** [comment]
Thanks @BurningTreeC

**Comment by twMat** [comment]
While this pull makes perfect sense, I'm just wondering if it would not prevent making tags from titles that do contain white space, considering how #3483 is not solved? I.e will my tiddler `Foo   ` be impossible to create a tag to/for?

**Comment by Jermolene** [comment]
Hi @twMat this PR doesn't affect the treatment of the text of tags themselves, it's just to make the wikitext more readable.

**Comment by BurningTreeC** [comment]
Hi @twMat @Jermolene ... On 5.1.21 it's possible to insert a tag like `Foo ` with a space at the end. On the current prerelease it's not possible anymore, so maybe this Pr was a mistake

**Comment by BurningTreeC** [comment]
Hi @twMat @Jermolene ... On 5.1.21 it's possible to insert a tag like `Foo ` with a space at the end. On the current prerelease it's not possible anymore, so maybe this Pr was a mistake

**Comment by pmario** [comment]
I think I did fix the spacing problem with: https://github.com/Jermolene/TiddlyWiki5/pull/4366 ... now it may be broken again.

**Comment by Jermolene** [comment]
I'm struggling to see how this PR could have affected whether tag with a trailing space can be created. I thought at some point we deliberately trimmed any whitespace?

Meanwhile, to your point in #4366 about copying and pasting, it may make more sense to use `&#32;` instead of `&nbsp;`.

**Pull request merged**